src/tide_provider/main.py

```diff
@@ -180,6 +180,39 @@ def parse_info(file_path: Path) -> list[TideInformation]:
     return infos
 
 
+def find_lowest_height_info_index(infos: list) -> int:
+    lowest = float(infos[0]["height"])
+    lowest_index = 0
+    for idx, info in enumerate(infos):
+        height = float(info["height"])
+        if height < lowest:
+            lowest = height
+            lowest_index = idx
+
+    return lowest_index
+
+
+def parse_canada_info(file_path: Path) -> list[TidalDataExtremum]:
+    infos: list[TidalDataExtremum] = []
+    with open(file_path, encoding="UTF-8") as f:
+        spamreader = csv.reader(f.readlines()[1:], delimiter=",")
+        for row in spamreader:
+            dt = datetime.strptime(
+                row[0],
+                "%Y-%m-%dT%H:%M:%S",
+            )
+            isotime = dt.astimezone(timezone.utc).isoformat()
+            height = row[1]
+            infos.append({"time": isotime, "height": height, "isHighTide": False})
+
+    is_high_tide = find_lowest_height_info_index(infos) % 2 == 1
```

Comment:
Interesting waz of doing this, but sure

Reply:
I'm not entirely happy with this either, it's the simplest + most reliable approach I could think of though.